commercetools.Sdk/IntegrationTests/commercetools.Sdk.IntegrationTests/GraphQL/GraphQLIntegrationTests.cs

```diff
@@ -38,5 +42,65 @@ public async Task CreateCustomerGroup()
                 Assert.Equal(category.Id, result.data.categories.results[0].id.Value);
             });
         }
+
+        [Fact]
+        public async Task UpdateAnOrder()
+        {
+            //Arrange
+            var orderNumber = $"order-{TestingUtility.RandomInt()}";
+            var query = @"mutation ($orderId:String, $orderNumber:String){
+                                updateOrder(
+                                             version: 1,
+                                             id:$orderId,
+                                             actions: [
+     	                                                {
+                                                            changeOrderState:{
+                                                              orderState: Confirmed
+                                                            }
+                                                        },
+                                                        {
+                                                            setOrderNumber:{
+                                                              orderNumber: $orderNumber
+                                                            }
+                                                        }
+                                                      ]
+                                            ) {
+                                                id
+                                              } 
+                                    }";
+            
+            //act
+            await WithCartWithSingleLineItem(client, 2, DefaultCartDraft,
+                async cart =>
+                {
+                    Assert.Single(cart.LineItems);
+                    
+                    //create order with null orderNumber
+                    await WithOrder(client, 
+                        draft => DefaultOrderFromCartDraftWithNumber(draft, cart, orderNumber: null),
+                        async order =>
+                        {
+                            Assert.NotNull(order);
+                            Assert.Null(order.OrderNumber);
+                            
+                            var queryParameters = new GraphQLParameters(query, new Dictionary<string, object>
+                            {
+                                {"orderId", order.Id},
+                                {"orderNumber", orderNumber}
+                            });
+                            var graphQlCommand = new GraphQLCommand<dynamic>(queryParameters);
+                            var result = await client.ExecuteAsync(graphQlCommand);
+                            Assert.NotNull(result);
```

Comment:
add an assertion which checks that the results `id` and `orderNumber` are the same as the once sent in the request. This could save the additional query to the API